src/Tests/Microsoft.DotNet.ApiCompatibility.Tests/Rules/CannotAddOrRemoveVirtualModifierTests.cs

```diff
@@ -0,0 +1,115 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.DotNet.ApiCompatibility.Abstractions;
+using Xunit;
+
+namespace Microsoft.DotNet.ApiCompatibility.Tests
+{
+    public class CannotAddOrRemoveVirtualModifierTests
+    {
+        [Fact]
+        public static void DifferencesReported()
+        {
+            string leftSyntax = @"
+namespace CompatTests
+{
+  public abstract class First {
+    public virtual void F() {}
+    public void G() {}
+    public abstract void H();
+  }
+}
+";
+            string rightSyntax = @"
+namespace CompatTests
+{
+  public abstract class First {
+    public void F() {}
+    public virtual void G() {}
+    public virtual void H() {}
+  }
+}
+";
+            IAssemblySymbol left = SymbolFactory.GetAssemblyFromSyntax(leftSyntax);
+            IAssemblySymbol right = SymbolFactory.GetAssemblyFromSyntax(rightSyntax);
+            ApiComparer differ = new();
+            IEnumerable<CompatDifference> differences = differ.GetDifferences(new[] { left }, new[] { right });
+            CompatDifference[] expected = new[]
+            {
+                new CompatDifference(DiagnosticIds.CannotRemoveVirtualModifier, string.Empty, DifferenceType.Removed, "M:CompatTests.First.F"),
+                new CompatDifference(DiagnosticIds.CannotAddVirtualModifier, string.Empty, DifferenceType.Added, "M:CompatTests.First.G"),
+            };
+            Assert.Equal(expected, differences);
+        }
+
+        [Fact]
+        public static void Null()
+        {
+            string leftSyntax = @"
+namespace CompatTests
+{
+  public class First {
+    public virtual void F() {}
+  }
+}
+";
+            string rightSyntax = @"
+namespace CompatTests
+{
+  public class First {
+    public virtual void G() {}
+  }
+}
+";
+            IAssemblySymbol left = SymbolFactory.GetAssemblyFromSyntax(leftSyntax);
+            IAssemblySymbol right = SymbolFactory.GetAssemblyFromSyntax(rightSyntax);
+            ApiComparer differ = new();
+            IEnumerable<CompatDifference> differences = differ.GetDifferences(new[] { left }, new[] { right });
+            Assert.NotEmpty(differences);
+        }
+
+        [Fact]
+        public static void StrictMode()
```

Comment:
Even when the names are long, you don't care about the length of the member because no one is going to call it, what you want is for it to be descriptive enough so that when it fails, you immediately know what the unit test is checking and even probably have a good guess of why it broke.

Reply:
I assumed the test runner would namespace these test names under `CannotAddOrRemoveVirtualModifierTests` inside `Microsoft.DotNet.ApiCompatibility.Tests`, so that I wouldn't have to guess. But I agree that this name could be more specific.

Reply:
It depends on which view you use, some do have the fully qualified name, but not all. Even then, I think it is fine to Omit from the name some things, like the name of the rule for example, since as you point out, this is already on the test class name, but having a more descriptive name of what the test is actually testing is useful.

Reply:
I usually follow our own documentation on unit test naming conventions: https://docs.microsoft.com/en-us/dotnet/core/testing/unit-testing-best-practices#naming-your-tests.

A method should include:
- The name of the method that is being tested
- The name of the scenario
- And the expected return/output

There's no need to encode the class name in it as it's already part of the method's signature.